src/transformers/tokenization_t5.py

```diff
@@ -78,6 +78,7 @@ class T5Tokenizer(PreTrainedTokenizer):
     def __init__(
         self,
         vocab_file,
+        bos_token="<s>",
```

Comment:
hi,
I think that this wouldn't work becouse `<s>` token does not exist in the original sentencepiece vocab (authors added only 3 special tokens - [link](https://github.com/google-research/text-to-text-transfer-transformer/blob/9fd7b14a769417be33bc6c850f9598764913c833/t5/data/sentencepiece_vocabulary.py#L35)). you could add `<s>` to the list of `additional_special_tokens` and then use `add_special_tokens` method to add it to encoder, but I think it is easier to use one of the existing extra_ids like shown below:
```suggestion
        bos_token="<extra_id_99>",
```

Reply:
Thank you for pointing out that. I agree with you that I shouldn't use '<s>' as the special token. In their original code, they actually use [ID 0](https://github.com/tensorflow/mesh/blob/b05b88c317ae78db6a678b3424fd24cc0982b1e2/mesh_tensorflow/transformer/transformer.py#L649) as the bos_token id. So I'm not sure if I should follow their way, or use another bos_token.

Reply:
If you want to use padding for autoregressive decoder, I think, the best is to use `pad_token` instead of `bos_token`. In the original code 0 is defined as pad token (please have a look at the link in my previous comment)